### PR TITLE
Faster rejection of irrelevant files

### PR DESCRIPTION
**Pull request by CorentinGrimaldi**
### Description

This pull request is to address #131 with a better file check relevancy by storing wmin and wmax in the cache file metadata. 
So we do not need to load all the cached file for nothing. Only the metadata are loaded and if the file is relevant for the calcul the file is loaded.
This allow for a much faster rejection of irrelevant file and speeding up the loading of large database such as CDSD-4000. Change from about 1s by file to 0.01s (approximately).

Fixes #<Issue Number>

### CODE REVIEW

File: radis/lbl/loader.py [hunk 1]
```diff
@@ -1966,36 +1927,37 @@ def load_and_concat(files, buffer):
                         cache=db_use_cached,
                         verbose=verbose,
                         drop_non_numeric=True,
+                        wavenum_min=wavenum_min,
+                        wavenum_max=wavenum_max,
                     )
+                    if isinstance(df, str):
+                        if df == "IrrelevantFile":
```
Comment (erwanp) [inline radis/lbl/loader.py:1934]: Could be replaced by a larger

```
try:
    df = .... 
except IrrelevantFileError:
    if db_use_cached=='force':
        raise
     else:
        continue
``` 

`df` is not rewritten.

**Review by erwanp — CHANGES_REQUESTED** [review]

**Comment by erwanp** [comment]
Will help #166 too

File: radis/io/cdsd.py [hunk 2]
```diff
@@ -116,7 +117,14 @@
...
 
 def cdsd2df(
-    fname, version="hitemp", count=-1, cache=False, verbose=True, drop_non_numeric=True
+    fname,
+    version="hitemp",
+    count=-1,
+    cache=False,
+    verbose=True,
+    drop_non_numeric=True,
+    wavenum_min=0.0,
```
Comment (erwanp) [inline radis/io/cdsd.py:126]: It's not straightforward what wavenum_min and max do here. Can you : 
- change the name to something explicit ? Ex : `load_only_wavenum_above=` ,  `load_only_wavenum_below=`? 
- add the names in the docstrings under Other Parameters with a shortdescription. In the docs, if needed, you can also refer to other functions like load_h5_cache_file using ``` :py:func`~radis.misc.cache_files` ```

**Review by erwanp — CHANGES_REQUESTED** [review]

File: radis/lbl/loader.py [hunk 3]
```diff
@@ -1898,104 +1899,60 @@ def load_and_concat(files, buffer):
...
-                        filename,
-                        cache=db_use_cached,
-                        verbose=verbose,
-                        drop_non_numeric=True,
-                    )
+                    except IrrelevantFileWarning:
+                        if db_use_cached=='force':
+                            raise
+                        else:
+                            continue
```
Comment (erwanp) [inline radis/lbl/loader.py:1952]: maybe add a low-level print here ? (if verbose==2 or 3  for instance ? )

File: radis/io/cdsd.py [hunk 4]
```diff
@@ -233,14 +250,19 @@ def cdsd2df(
...
+                cache,
+                metadata=metadata,
+                current_version=radis.__version__,
+                last_compatible_version=OLDEST_COMPATIBLE_VERSION,
+                verbose=verbose,
+                load_only_wavenum_above=load_only_wavenum_above,
+                load_only_wavenum_below=load_only_wavenum_below,
+            )
+        except IrrelevantFileWarning as err:
+            raise err
```
[the remark this answers is not in the document]
Comment (CorentinGrimaldi) [reply radis/io/cdsd.py:265]: Ah yes, no need to do that, in my mind it was a way to transmit the warning to the upper function.
Will remove it.

[the remark this answers is not in the document]
Comment (CorentinGrimaldi) [reply radis/io/cdsd.py:265]: As in radis/io/hitran.py

**Review by erwanp — APPROVED** [review]

**Comment by erwanp** [comment]
Very nice, merging ! 

@BlehMaks have a look at the PR, it can be useful to you : you may need to regenerate your cache files, but then it will make the whole loading process super fast even with the whole CDSD-4000 Database !

**Pull request merged**